### PR TITLE
Create battle

### PR DESCRIPTION
**Pull request by pamella**
## Description

- [Create a new battle](https://trello.com/c/jDv2Dda0/1-as-a-trainer-i-create-a-new-battle)
- [Trainer opponent able to select its pokemon team](https://trello.com/c/zjdcC54A/4-as-the-opponent-trainer-i-select-my-pkns-to-battle)
- [Message error for pokemon team sum more than 600 points](https://trello.com/c/1Ba33KIb/17-as-a-trainer-i-see-form-errors-in-case-the-pkns-i-select-sum-more-than-600-points)
- Simple list of on going and settled battles
- Feedback messages after submitting battle forms.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires documentation updates.
- [ ] I have updated the documentation accordingly.
- [X] My change requires dependencies updates.
- [X] I have updated the dependencies accordingly.

### CODE REVIEW

File: battles/forms.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        self.fields['trainer_creator'].widget = forms.HiddenInput()
+        self.fields['status'].initial = 'ON_GOING'
+        self.fields['status'].widget = forms.HiddenInput()
+
+        pkm1 = forms.CharField()
+        pkm2 = forms.CharField()
+        pkm3 = forms.CharField()
+        self.fields['pokemon_1'] = pkm1
+        self.fields['pokemon_2'] = pkm2
+        self.fields['pokemon_3'] = pkm3
```
Comment (hugobessa) [inline battles/forms.py]: you could declare these fields' widgets in Meta.widgets attribute instead of the constructor, I think it would be easier to read.

And about the initial values it would be better not to override them indefinitely, but only when no initial is passed on the form initialization. Something like that:

```python
pokemon_1 = forms.CharField()
pokemon_2 = forms.CharField()
pokemon_3 = forms.CharField()

class Meta:
    model = Battle
    fields = ('trainer_opponent', 'trainer_creator', 'status')
    widgets = {
        "trainer_creator": forms.HiddenInput(),
        "status": forms.HiddenInput(),
    },

def __init__(self, *args, **kwargs):
    initial = kwargs.get('initial', {})
    initial_user_id = initial.get('user') and initial.get('user').id 
    initial['trainer_creator'] = initial.get('trainer_creator', initial_user_id)
    initial['status'] = initial.get('status', 'ON_GOING')
    kwargs['initial'] = initial
    super().__init__(*args, **kwargs)

File: battles/forms.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        return super().save()
+
+
+class SelectTrainerTeamForm(forms.ModelForm):
+    class Meta:
+        model = TrainerTeam
+        fields = ('trainer', 'battle_related')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (hugobessa) [inline battles/forms.py]: You can use the same logic I showed in the comment in the form above

File: battles/views.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+from battles.forms import CreateBattleForm, SelectTrainerTeamForm
+from battles.models import Battle, TrainerTeam
+
+
+class CreateBattleView(
+    LoginRequiredMixin,
+    SuccessMessageMixin,
+    CreateView
+):
```
Comment (hugobessa) [inline battles/views.py]: I don't bother with this, but just to let you know if you have a future reviewer that bothers... We usually  define class multiline signatures like this:
```python
class CreateBattleView(
        LoginRequiredMixin, SuccessMessageMixin, CreateView, LoginRequiredMixin, 
        SuccessMessageMixin, CreateView):  
    pass
```

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,102 @@
+amqp==2.4.1
```
Comment (hugobessa) [inline requirements.txt]: why is this file necessary?

File: Pipfile [hunk 5]
```diff
@@ -35,6 +36,11 @@ whitenoise = "*"
 psutil = "*"
 ipython = "*"
 psycopg2-binary = "*"
+django-widget-tweaks = "*"
+requests = "*"
+ipdb = "*"
+pylint = "==2.1.1"
+astroid = "==2.0.4"
```
[the remark this answers is not in the document]
Comment (pamella) [reply Pipfile]: Done!

File: requirements.txt (hunk 4 above)
Comment (pamella) [reply requirements.txt]: I pushed when I was using pip instead of pipenv at home, but now it's deleted from the repository.

File: battles/views.py (hunk 3 above)
Comment (pamella) [reply battles/views.py]: Done!

File: battles/forms.py (hunk 1 above)
Comment (pamella) [reply battles/forms.py]: I didn't know I could do this way. Thank you!

File: battles/forms.py (hunk 2 above)
Comment (pamella) [reply battles/forms.py]: Done!

**Comment by pamella** [comment]
> I see you didn't write tests on this PR. Is that something you arranged with Rob? If not let's try to add some unit/integration tests here.

@hugobessa about this, I talked to Rob and Im gonna start to learn about test in order to write tests for those features.
